packages/cli/src/types/CliConfig.ts

```diff
@@ -0,0 +1,47 @@
+import { ColumnDefProperties } from '@synthql/queries';
+export interface CliConfig {
+    /**
+     * Internal. DO NOT USE/SET this option
+     * Specifies which draft of the JSON Schema standard the schema adheres to
+     */
+    $schema?: string;
+    /**
+     * The output directory for the generated files
+     * e.g. `'src/generated'`
+     */
+    out: string;
+    /**
+     * The default schema to use e.g. `'public'`
+     * This is similar to the `search_path` in PostgreSQL
+     */
+    defaultSchema: string;
+    /**
+     * The schemas to include in generation e.g. `['public']`
+     */
+    schemas: Array<string>;
+    /**
+     * The tables to include in generation e.g. `['users', 'accounts']`
+     */
+    tables?: Array<string>;
+    /**
+     * The table definitions for any tables you want
+     * to be overriden during the schema generation
+     */
+    schemaDefOverrides?: SchemaDefOverrides;
+}
+export interface SchemaDefOverrides {
+    /**
+     * The column definitions for each table to be overriden
+     * The key for each table should be the qualified name for the table,
+     * in the form `${schemaName.tableName}` e.g `'public.users'`
+     */
+    [x: string]: TableDefOverrides | undefined;
```

Comment:
I would suggest we don't introduce `TableDefOverrides`. Also that we give `x` a better name.

```suggestion
    [table: string]: { [column: string]: Partial<ColumnDefProperties> }
```